composeApp/src/androidMain/kotlin/org/ooni/probe/MainActivity.kt

```diff
@@ -50,7 +51,23 @@ class MainActivity : ComponentActivity() {
     }
 
     private fun manageIntent(intent: Intent) {
-        if (intent.action != Intent.ACTION_VIEW) return
+        when (intent.action) {
+            Intent.ACTION_VIEW -> manageOoniRun(intent)
+            Intent.ACTION_SEND -> manageSend(intent)
+            else -> return
+        }
+    }
+
+    private fun manageSend(intent: Intent) {
+        val url = intent.getStringExtra(Intent.EXTRA_TEXT) ?: return
+        if (Patterns.WEB_URL.matcher(url).matches()) {
```

Comment:
It would have been better to just let the ChooseWebsites handle the url, and validate if it's a good URL. It uses a different validation logic than `Patterns.WEB_URL`. `Deeplink.Error` is also not a very clear name.